mowgli_etl/pipeline/wdc/wdc_offers_corpus.py

```diff
@@ -4,16 +4,41 @@
 
 
 class WdcOffersCorpus:
+    """
+    Container for data source used to generate and access data entries easily
+
+    :param wdc_json_file_path: directory of data source
+    """
+
     def __init__(self, *, wdc_json_file_path: Path):
+        """
+        Constructor method
+        """
+
         self.__file_path = wdc_json_file_path
-        self.__file_length = open(self.__file_path).read().count("\n") + 1
+        with open(self.__file_path) as data:
+            self.__file_length = data.read().count('\n') + 1
```

Comment:
Actually, this is kind of self-defeating, since it's going to buffer the whole file in memory, albeit temporarily. This should be a line iteration.